### PR TITLE
feat: add a job to initialize mongodb users

### PR DESCRIPTION
**Pull request by MoisesGSalas**
# Description
Include an initialization job similar to the MySQL one that creates a MongoDB user with the necessary permissions. To simplify things a bit we use the same user for edxapp and forum and remove the need for the forum-overrides patch. This assumes an admin user is already present on your MongoDB instance and can be accessed via `MONGODB_ROOT_USERNAME` and `MONGODB_ROOT_PASSWORD`.

### CODE REVIEW

**Comment by Henrrypg** [comment]
Can we add the [BACKUP](https://github.com/eduNEXT/drydock-backups/blob/main/drydock_backups/plugin.py#L31-L32) user?

**Comment by Henrrypg** [comment]
It works with [backup](https://www.mongodb.com/docs/manual/reference/built-in-roles/#mongodb-authrole-backup) role

**Review by Ian2012 — APPROVED** [review]

**Comment by MoisesGSalas** [comment]
@Henrrypg, I think the creation of the backup user might be better handled by the backup plugin.

We could do two things: we can define almost the same job in the backup plugin. or we can define a new patch to add additional instructions to the job.

I think the patch might be a better fit.

**Review by Henrrypg — APPROVED** [review]

File: drydock/plugin.py [hunk 1]
```diff
@@ -49,6 +49,12 @@
 }
 
 hooks.Filters.CONFIG_DEFAULTS.add_items([("OPENEDX_DEBUG_COOKIE", "ednx_enable_debug")])
+hooks.Filters.CONFIG_OVERRIDES.add_items([
+    # This values are not prefixed with DRYDOCK_
+    ("MONGODB_ROOT_USERNAME", "admin"),
```
[the remark this answers is not in the document]
Comment (MoisesGSalas) [reply drydock/plugin.py:48]: Yes, the `CONFIG_OVERRIDES` filter will simply add the entries with the default values to the file if they are missing for more visibility,

File: drydock/templates/drydock/k8s/drydock-jobs/mongodb.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: mongodb
+        image: {{ DOCKER_IMAGE_MONGODB }}
+        command: ["/bin/sh", "-c"]
+        args:
+        - |
+          echo "Initialising MongoDB..."
+          mongo --host {{MONGODB_HOST }} -u {{ MONGODB_ROOT_USERNAME }} -p {{ MONGODB_ROOT_PASSWORD }} admin <<EOF
```
[the remark this answers is not in the document]
Comment (MoisesGSalas) [reply drydock/templates/drydock/k8s/drydock-jobs/mongodb.yml]: I've changed the default values of `MONGODB_ROOT_USERNAME` and `MONGODB_ROOT_PASSWORD` to be empty and added an additional pair of quotes to the command so it doesn't fail with the default mongo that comes with tutor without authentication.

**Review by jfavellar90 — APPROVED** [review]

**Comment by jfavellar90** [comment]
I want to backport this to Olive. However, we must adjust the release workflow since it does not work properly. @Henrrypg could we do a manual release? the branch to apply the changes to is called `olmo-release`.

**Comment by Henrrypg** [comment]
@jfavellar90 backported

**Pull request merged**